ambry-coordinator/src/test/java/com.github.ambry.coordinator/MockBlockingChannel.java

```diff
@@ -152,13 +156,14 @@ public void send(Send send)
             break;
 
           case Blob:
-            MockDataNode.BlobOutputAndError bdae = mockDataNode.getData(blobId);
+            MockDataNode.BlobDataAndError bdae = mockDataNode.getData(blobId);
             getResponseErrorCode = bdae.getError();
             if (getResponseErrorCode == ServerErrorCode.No_Error) {
-              BlobOutput blobData = bdae.getBlobOutput();
-              byteBufferSize = (int) MessageFormatRecord.Blob_Format_V1.getBlobRecordSize(blobData.getSize());
+              BlobData blobData = bdae.getBlobData();
+              byteBufferSize = (int) MessageFormatRecord.Blob_Format_V2.getBlobRecordSize(blobData.getSize());
               byteBuffer = ByteBuffer.allocate(byteBufferSize);
-              MessageFormatRecord.Blob_Format_V1.serializePartialBlobRecord(byteBuffer, blobData.getSize());
+              MessageFormatRecord.Blob_Format_V2
```

Comment:
Nope. We need to know the BlobType. So BlobData is noting but (all contents of BlobLOutput) along with BlobType. So ideally it should be size, InputStream and BlobType.